engine/engine.go

```diff
@@ -44,13 +48,18 @@ func NewEngineAPI(
 	b *builder.Builder,
 	txValidator TxValidator,
 	blockStore DB,
+	appChainClient *appchainClient.Context,
 	metrics Metrics,
 ) *EngineAPI {
+	privKey := ed25519.GenPrivKeyFromSecret([]byte("monomer")) // TODO: configurable
```

Comment:
Will the `privKey` secret eventually be an environmental variable that's configured when monomer starts up? Or will it just be a random private key?

Reply:
I think there could be use cases in either direction, so we'll probably want it consumable as an environment variable / command flag / whatever.

One might want to:
- sign deposit txs with a secret key (providing an authentication on the data)
- sign deposit txs with a known key (which makes them compatible with the current workings of appchain integrations, but provides no authentication)
- not sign deposit txs (not currently working, but could work in future with AnteHandler modifications or some modification on a custom tx type)